src/BuildScriptGenerator/BuildScriptGeneratorServiceCollectionExtensions.cs

```diff
@@ -31,7 +31,6 @@ public static IServiceCollection AddBuildScriptGeneratorServices(this IServiceCo
             services.AddSingleton<ISourceRepoProvider, DefaultSourceRepoProvider>();
             services.AddSingleton<ITempDirectoryProvider, DefaulTempDirectoryProvider>();
             services.AddSingleton<IScriptExecutor, DefaultScriptExecutor>();
-            services.AddSingleton<IEnvironmentSettingsProvider, DefaultEnvironmentSettingsProvider>();
```

Comment:
- This feature was implemented during our initial days and was not advertised.
- We left the source code in to use it later if needed.
- I am sending this PR now because I will later be sending a PR where we would be following `Microsoft.Extensions.Configuration` api to get hierarchical configuration from different 'sources' (command line -> environment variables -> files like `build.env` and in future maybe `app.yml`)

Reply:
So do we also want the same hierarchical config for run-time command line and env variables as well in the future?

Reply:
yes, that's right.